tests/core/pthread/emscripten_atomics.c

```diff
@@ -0,0 +1,38 @@
+#include <assert.h>
+#include <emscripten/threading.h>
+#include <stdio.h>
+#include <stdint.h>
+#include <string.h>
+
+#define OLD 42
+
+int main() {
+  char buffer[16];
+  memset(&buffer[0], OLD, sizeof(buffer));
+
+  uint8_t o8 = emscripten_atomic_exchange_u8(&buffer[0], 1);
+  assert(o8 == OLD);
+  assert(buffer[0] == 1);
+  assert(buffer[1] == OLD);
+
+  uint16_t o16 = emscripten_atomic_exchange_u16(&buffer[0], 2);
+  assert((o16 & 0xff) == 1);
+  assert((o16 >> 8) == OLD);
+  assert(buffer[0] == 2);
+  assert(buffer[1] == 0);
+  assert(buffer[2] == OLD);
+
+  uint16_t o32 = emscripten_atomic_exchange_u32(&buffer[0], 3);
```

Comment:
Should this be a `uint32_t`?

Reply:
Thanks!

Also I realized we should check both the pthreads stubs and the full pthreads version, I added that now too.